G1ANT.Addon.Net/ImapHelper.cs

```diff
@@ -7,19 +7,41 @@ namespace G1ANT.Addon.Net
 {
     public static class ImapHelper
```

Comment:
Make it non-static `ImapConnectionManager` or `ImapManager`. You can make it a singleton (blah).